python-package/lightgbm/basic.py

```diff
@@ -574,7 +586,8 @@ def predict(self, data, num_iteration=-1,
             preds, nrow = self.__pred_for_csr(csr, num_iteration, predict_type)
         if pred_leaf:
             preds = preds.astype(np.int32)
-        if is_reshape and preds.size != nrow:
+        is_sparse = scipy.sparse.issparse(preds) or isinstance(preds, list)
```

Comment:
Should `predict_type` be checked here? If I'm not mistaken, for all other predictions other than `C_API_PREDICT_CONTRIB` we still need to reshape result.

Reply:
no, it is not necessary, since all other predictions will be dense, only C_API_PREDICT_CONTRIB for sparse input can output a sparse output

Reply:
Yeah, that's right! But only for now. Anyway, we should keep in mind that future contributions may require updating this condition.